django/core/serializers/xml_serializer.py

```diff
@@ -56,7 +56,7 @@ def start_object(self, obj):
         if not self.use_natural_primary_keys or not hasattr(obj, "natural_key"):
             obj_pk = obj.pk
             if obj_pk is not None:
-                attrs["pk"] = str(obj_pk)
+                attrs["pk"] = self._handle_pk_field(obj)
```

Comment:
I think this should be `obj._meta.value_to_string` and that we're missing a `CompositeField.value_from_object`?

Reply:
Thank you! I made some updates